ppoprf/src/ppoprf.rs

```diff
@@ -37,6 +37,7 @@ use crate::{ggm::GGM, PPRF};
 pub const COMPRESSED_POINT_LEN: usize = 32;
 pub const DIGEST_LEN: usize = 64;
 pub const MAX_SERIALIZED_PK_SIZE: usize = 16384;
+pub const MAX_SERIALIZED_PROOF_SIZE: usize = 32768;
```

Comment:
Shouldn't this be `2 * 32` since the struct is only two scalars?

Reply:
yeah, probably. how about 512? a little bit of padding doesn't hurt

Reply:
I'd prefer an accurate bound. Someone could use this for pre-allocation, and tests should catch if it's too small.

Reply:
how about 128 then? i think having a small amount of allowance would be good if we wanted to add something to the proof, or if the bincode format changes. base 2 numbers are nice to look at as well.

Reply:
> What changes are you worried about leaving room for?

if bincode decides to add an extra couple bytes of metadata in the future, or if we decide to add an extra field for whatever reason

> Doesn't that make it easy to adjust the max serialized size if it anything changes?

yes, that's true, but it does remove the extra 'gotcha' if we need to add an extra field or update to a breaking version of bincode, and our tests are failing.

imo, the extra 64 bytes doesn't seem like a biggie and should protect against the vast majority of memory violation cases

Reply:
Unfortunately I don't think there's a way to have the compiler calculate this, even if Rust had richer constexpr support. OTOH, having it manually calculated and reviewed can catch errors if something changes inadvertently. That's the main reason I'd like it to be a tight bound.

It's also a public constant. Whether doubling the space requirement is an issue or not depend on what the caller is doing. What if it's trying to keep ten thousand of them on the stack? Please just make this `2 * 32`.

Reply:
> OTOH, having it manually calculated and reviewed can catch errors if something changes inadvertently. That's the main reason I'd like it to be a tight bound.

That's what snapshot testing via `insta` is for. If changes are detected in the output, the developer is forced to review the changes and verify that they are desired. The max size is for protecting against extreme cases of memory violations.

> What if it's trying to keep ten thousand of them on the stack?

Could be wrong, but sounds like an unlikely scenario. If it isn't, 128-byte proofs would consume a little over 1 meg.

nevertheless, the change has been made, hopefully we can merge soon.